### DISCUSSION TITLE
How to rename a table

### DISCUSSION THREAD

**Discussion by felinto-dev** [prisma/prisma#8557] [Q&A] [answered]
How could I rename a table using Prisma in a safe way? (without data loss)

**Comment by ryands17** [comment, marked as answer]
@felinto-dev 👋 

You can perform this in the following steps:

1. Edit the model name in your `schema.prisma`
2. Run `prisma migrate dev --create-only`. A migration `.sql` file will be generated
3. Edit that file to replace the drop table commands with `alter table` 
4. Run `prisma migrate dev`

This will make sure there's no data loss after updating a table name.

**Reply by adamzerner** [reply]
This doesn't work for me. Perhaps Prisma Migrate has changed since this answer was posted? At step 2 I get the following:

```
code/personal-dashboard-api [master●] » npx prisma migrate dev  --create-only
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "personal_dashboard_api_development", schema "public" at "localhost:5432"


⚠️  Warnings for the current datasource:

  • You are about to drop the `CustomFilter` table, which is not empty (1 rows).

? Are you sure you want create this migration? › (y/N)
```

If I select `y` I lose data, which I'd like to avoid. It happens immediately without giving me a chance to intervene. If I select `N` I'm unable to proceed past step 2.

And it is the same thing if I try to change a column name rather than a table name. Here is the output I get when trying to change a column name that has data:

```
code/personal-dashboard-api [master●] » npx prisma migrate dev  --create-only
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "personal_dashboard_api_development", schema "public" at "localhost:5432"



⚠️ We found changes that cannot be executed:

  • Step 0 Added the required column `nameTest` to the `CustomFilter` table without a default value. There are 1 rows in this table, it is not possible to execute this step.


⚠️  Warnings for the current datasource:

  • You are about to drop the column `name` on the `CustomFilter` table, which still contains 1 non-null values.

? Are you sure you want create this migration? › (y/N)
```

Here is the relevant model in the schema:

```
model CustomFilter {
  id        Int      @id @default(autoincrement())
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  nameTest  String
  filters   Json
  user      User     @relation(fields: [userId], references: [id])
  userId    Int
}
```

But interestingly, the [relevant docs](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/customizing-migrations#renaming-a-field=) say the following:

> To actually rename a field and avoid data loss when you run the migration in production

Note the "in production" part. The output I posted above is from development. Is it the case that I'd lose data in development only and _not_ in production? How certain is this? Losing data in production would be really bad, so I'd like to be extremely confident before proceeding.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by Tobbe** [comment]
For anyone coming here in the future, I detailed my workaround here https://github.com/prisma/prisma/issues/12387#issuecomment-3036271110

It avoids the scary warning that @ryands17's suggested solution will give you (as pointed out by @adamzerner)
> ```
> ⚠️  Warnings for the current datasource:
> 
>   • You are about to drop the column `name` on the `TableName` table, which still contains 100 non-null values.
> 
> ? Are you sure you want create this migration? › (y/N)
> ```

**Reply by EduardoBautista** [reply]
@Tobbe This thread is about renaming a table, not a column.

**Reply by Tobbe** [reply]
@EduardoBautista  Yes. That's what I'm doing in the comment I liked to, right?
